docs/SONATA_DEVELOPER_GUIDE.md

```diff
@@ -607,39 +607,39 @@ Table 1: The Layout  of the HDF5 file format for describing nodes.
 
 ##### Nodes - Required Attributes
 
-**/nodes** - The top-level group that will contain populations.  This group must always be present. 
+**/nodes** - The top-level group that will contain populations.  This group must always be present.
 
 **/nodes/<population_name>** - defines a node population with the given name, which is also the node population name that will be used in edge files to specify the source or target node populations of an edge population (see below).  Multiple populations may be defined in one HDF5 file.
 
-**node_type_id** -  This is a unique integer for every node type used to associate a node to a node type.  A node type has associated attributes, and a node inherits attributes from its node type .  Attributes associated with a node override attributes inherited from the node type CSV. node_type_id is a unique integer associated with each node type and is used to index all the node type properties associated with a given node with known node_type_id.  These need not be ordered or contiguous, but must be unique.  A reference implementation might be to use a key-value store, such as a dictionary to associate a node_type_id with its associated column values.
+**node_type_id** [INT32] -  This is a unique integer for every node type used to associate a node to a node type. A node type has associated attributes, and a node inherits attributes from its node type. Attributes associated with a node override attributes inherited from the node type CSV. node_type_id is a unique non-negative integer associated with each node type and is used to index all the node type properties associated with a given node with known node_type_id.  These need not be ordered or contiguous, but must be unique. If no base node type is used for a particular node, `-1` value is used. A reference implementation might be to use a key-value store, such as a dictionary to associate a node_type_id with its associated column values.
```

Comment:
Allowing for no node_types may be a potentially big deal for us. We may agree to it, but we need time to talk it over on our end. For now you can just set them to 0 and have a node_types_table with only a node_type_id column. Not ideal but at this point I'm not keen on making big changes to the requirements.